dotnet/src/Connectors/Connectors.Memory.Pinecone/PineconeClient.cs

```diff
@@ -330,15 +325,15 @@ public async Task UpdateAsync(string indexName, PineconeDocument document, strin
             .WithFilter(filter)
             .Build();
 
-        (HttpResponseMessage response, string responseContent) = await this.ExecuteHttpRequestAsync(basePath, request, cancellationToken).ConfigureAwait(false);
+        string? responseContent = null;
 
         try
         {
-            response.EnsureSuccessStatusCode();
+            (_, responseContent) = await this.ExecuteHttpRequestAsync(basePath, request, cancellationToken).ConfigureAwait(false);
         }
-        catch (HttpRequestException e)
+        catch (HttpOperationException e)
         {
-            this._logger.LogDebug("Index not found {0}", e.Message);
+            this._logger.LogDebug("Index not found {Message}", e.Message);
```

Comment:
Same here, in the same class the log level is inconsistent. If you already updated string template in these places, maybe it's worth to update log level as well. It can be also done in separate PR.

Reply:
Easy fix, done.